LeetCodeSolutions/KeysAndRooms/KeysAndRooms.cs

```diff
@@ -0,0 +1,49 @@
+namespace LeetCodeSolutions;
+public class KeysAndRoomsSolution
+{
+
+
+    private class Rooms
+    {
+        private IList<IList<int>> _rooms;
+        public Rooms(IList<IList<int>> rooms)
+        {
+            _rooms = rooms;
+        }
+
+        private Boolean AllRoomsTraversed(bool[] visited)
+        {
+            foreach (bool room in visited)
+            {
+                if (room) continue;
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private Boolean DFSUtil(bool[] visited, int roomNumber = 0)
```

Comment:
I'm pretty sure this needs to save the state of the visited array to track the paths, but worth trying out iteratively